types/node/fs.d.ts

```diff
@@ -579,12 +579,17 @@ declare module "fs" {
      * @param fd A file descriptor.
      */
     export function fstatSync(fd: number): Stats;
+        
+    export type lstatOptions = {
+        bigint?: boolean;
+    };
 
     /**
      * Asynchronous lstat(2) - Get file status. Does not dereference symbolic links.
      * @param path A path to a file. If a URL is provided, it must use the `file:` protocol.
      */
     export function lstat(path: PathLike, callback: (err: NodeJS.ErrnoException | null, stats: Stats) => void): void;
+    export function lstat(path: PathLike, options: lstatOptions, callback: (err: NodeJS.ErrnoException | null, stats: Stats) => void): void;
```

Comment:
Even if you request `bigint` result the type definition here will use `Stats<number>`.
Take a look at `stat` which has two overloads regarding this, one with `bigint: true` returning `Stats<bigint>` and one with a boolean returning both types. I think the same is needed here.
And I think you should also update the `__promisify__` variants for the `lstat` namespace.

And as usual tests would be good.